### PR TITLE
Implement xDS circuit breaking interop test (advanced)

### PR DESCRIPTION
**Pull request by voidzcy**
Enhanced xDS circuit breaking interop test. See updated go/grpc-xds-circuit-breaking-interop-tests.

For languages wanting to enable circuit breaking test, add circuit_breaking to the --test_case option and use the new server image projects/grpc-testing/global/images/xds-test-server-3.

--------------
Requires proto definition change in #24703 
Replacement of #24696

### CODE REVIEW

File: tools/run_tests/run_xds_tests.py [hunk 1]
```diff
@@ -357,6 +408,62 @@ def wait_until_all_rpcs_go_to_given_backends(backends,
...
+      num_rpcs: Expected number of RPCs to be in-flight.
+      threshold: Number within [0,100], the tolerable percentage by which
+        the actual number of RPCs in-flight can differ from the expected number.
+    '''
+    if threshold < 0 or threshold > 100:
+        raise ValueError('Value error: Threshold should be between 0 to 100')
+    threshold_fraction = threshold / 100.0
+    start_time = time.time()
+    error_msg = None
+    logger.debug('Waiting for %d sec until %d RPCs (with %d%% tolerance) in-flight'
```
Comment (menghanl) [inline tools/run_tests/run_xds_tests.py:429]: Print the RPC type
`"until %d %s RPCs" ... rpc_type`

File: tools/run_tests/run_xds_tests.py [hunk 2]
```diff
@@ -357,6 +408,62 @@ def wait_until_all_rpcs_go_to_given_backends(backends,
...
+            break
+    # Ensure the number of outstanding RPCs is stable.
+    if not error_msg:
+        time.sleep(5)
+        error_msg = _check_rpcs_in_flight(rpc_type,
+                                          num_rpcs,
+                                          threshold,
+                                          threshold_fraction)
+    if error_msg:
+        raise Exception(error_msg)
```
Comment (menghanl) [inline tools/run_tests/run_xds_tests.py:448]: This error string is just `actual(%d) < expected(%d - %d%%)`. Add detailed context.

File: tools/run_tests/run_xds_tests.py [hunk 3]
```diff
@@ -1207,7 +1417,6 @@ def create_target_proxy(gcp, name):
         config = {
             'name': name,
             'url_map': gcp.url_map.url,
-            'validate_for_proxyless': True,
```
Comment (menghanl) [inline tools/run_tests/run_xds_tests.py:1210]: I don't think we want to disable `validate_for_proxyless` for all the tests. We should only disable for the new test.

I had to do the same for the path_matching tests.
See https://github.com/grpc/grpc/commit/fc87ccb252fe0d5dd0bba18b4585c0e4068dd2c8#diff-4ca3acdaf94b6dd59a2d7da9596e824ca061dbdf0b6a747fdadf98e1c3785c42L1197 (where it was removed).

At that time, patching target_grpc_proxy didn't work. Not sure if that's the same today.

Comment (voidzcy) [reply tools/run_tests/run_xds_tests.py:1210]: Fixed. Borrowing your code.

File: tools/run_tests/run_xds_tests.py (hunk 2 above)
Comment (voidzcy) [reply tools/run_tests/run_xds_tests.py]: Enhanced.

File: tools/run_tests/run_xds_tests.py (hunk 1 above)
Comment (voidzcy) [reply tools/run_tests/run_xds_tests.py]: Enhanced.

**Review by ericgribkoff — APPROVED** [review]

**Pull request merged**